test/torchaudio_unittest/functional/torchscript_consistency_impl.py

```diff
@@ -617,6 +617,28 @@ def test_phase_vocoder(self):
         )[..., None]
         self._assert_consistency_complex(F.phase_vocoder, (tensor, rate, phase_advance))
 
+    def test_mvdr_weights_souden(self):
+        def func(psd_speech, psd_noise, reference_channel: int):
```

Comment:
I presume you needed this helper function to make it work with TorchScript-ing Union type. 
That should not be the case. The current code does not seem to perform type refinement on Tensor type. Try using ‘is instance’ on the second clause to see if you can get rid of the helper function.